tests/test_scylla_repository.py

```diff
@@ -115,6 +118,62 @@ def test_setup_unstable_master_new_url(self):
+        file_text = source_file.read_text()
+        file_text = file_text.replace("hash=", "hash=123")
+        source_file.write_text(file_text)
+
+    def test_setup_unstable_master_no_unified_package_reinstall(self):
+        """
+        Validate that if package hash is changed, new package will be downloaded.
+        - download the Scylla packages. Packages hash will be saved in the "source.txt" file under relevant package folder
+        - change the hash to be wrong for one of the packages (choose the package randomly). No matter hash of which package is wrong -
+        all packages should be re-downloaded
+        - run setup again. It expected that the packages will be downloaded again. The download time should be not short.
+        Actually time without download should be around 5 ms, and with download about 35 ms. I put here more than 20
+        """
+        cdir, version = scylla_setup(version="unstable/master:2021-01-18T15:48:13Z", verbose=True, skip_downloads=False)
+        assert '2021-01-18T15_48_13Z' in cdir
+        assert version == '4.4.dev'
+
+        package_to_corrupt = random.choice([CORE_PACKAGE_DIR_NAME, "scylla-tools-java", "scylla-jmx"])
+        self.corrupt_hash_value(Path(cdir) / package_to_corrupt / SOURCE_FILE_NAME)
+
+        start_time = time.time()
+        cdir, version = scylla_setup(version="unstable/master:2021-01-18T15:48:13Z", verbose=True, skip_downloads=False)
+        end_time = time.time()
+        assert (end_time - start_time) > 20
+
+        assert '2021-01-18T15_48_13Z' in cdir
+        assert version == '4.4.dev'
+
+    def test_setup_unstable_master_unified_package_reinstall(self):
+        """
+        Validate that if package hash is changed, new package will be downloaded.
+        - download the unified package. Package hash will be saved in the "source.txt" file
+        - change the hash to be wrong
+        - run setup again. It expected that the package will be downloaded again. The download time should be not short.
+        Actually time without download should be less than 3 ms, and with download about 9 ms. I put here more than 20
+        """
+        cdir, version = scylla_setup(version="unstable/master:2023-04-03T22:38:18Z", verbose=True, skip_downloads=False)
+        assert '2023-04-03T22_38_18Z' in cdir
+        assert version == '5.3.0-dev'
+
+        self.corrupt_hash_value(Path(cdir) / CORE_PACKAGE_DIR_NAME / SOURCE_FILE_NAME)
+
+        start_time = time.time()
+        cdir, version = scylla_setup(version="unstable/master:2023-04-03T22:38:18Z", verbose=True, skip_downloads=False)
+        end_time = time.time()
+        assert (end_time - start_time) > 5
+        assert '2023-04-03T22_38_18Z' in cdir
+        assert version == '5.3.0-dev'
+
+    def test_get_local_tarball_hash(self):
+        this_path = Path().resolve()
+        url_hash = get_url_hash(url=str(this_path / "tests" / "test_data" / "scylla_unified_master_2023_04_03.tar.gz"))
```

Comment:
where is that file ? in the the source code ? 

let make sure it's not a full sized release, the a minimal empty `tar.gz`

Reply:
and yeah, CI show there no such file:
```
FAILED tests/test_scylla_repository.py::TestScyllaRepositoryRelease::test_get_local_tarball_hash - requests.exceptions.MissingSchema: Invalid URL '/home/runner/work/scylla-ccm/scylla-ccm/tests/test_data/scylla_unified_master_2023_04_03.tar.gz': No scheme supplied. Perhaps you meant https:///home/runner/work/scylla-ccm/scylla-ccm/tests/test_data/scylla_unified_master_2023_04_03.tar.gz?
```

Reply:
I took is from unstable. It's size 166M

Reply:
> and yeah, CI show there no such file:
> 
> ```
> FAILED tests/test_scylla_repository.py::TestScyllaRepositoryRelease::test_get_local_tarball_hash - requests.exceptions.MissingSchema: Invalid URL '/home/runner/work/scylla-ccm/scylla-ccm/tests/test_data/scylla_unified_master_2023_04_03.tar.gz': No scheme supplied. Perhaps you meant https:///home/runner/work/scylla-ccm/scylla-ccm/tests/test_data/scylla_unified_master_2023_04_03.tar.gz?
> ```

Yeah, I did not added it. fixing

Reply:
@fruch 
Oh, it is too big. It's is allowed 100m. I'll take any tarball. It is not must to be scylla package, right?

Reply:
fixed